PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['res.company']
-    _inherit = ['res.config.settings']
-    _inherit = ['res.partner']

Total Changes: 10

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/partner_autocomplete/models/iap_autocomplete_api.py
+++ b/addons/partner_autocomplete/models/iap_autocomplete_api.py
@@ -11,6 +11,7 @@ _logger = logging.getLogger(__name__)
 
 
 class IapAutocompleteApi(models.AbstractModel):
+    _name = 'iap.autocomplete.api'
     _description = 'IAP Partner Autocomplete API'
     _DEFAULT_ENDPOINT = 'https://partner-autocomplete.odoo.com'
 

--- a/addons/partner_autocomplete/models/ir_http.py
+++ b/addons/partner_autocomplete/models/ir_http.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class IrHttp(models.AbstractModel):
-    _inherit = ['ir.http']
+    _inherit = 'ir.http'
 
     def session_info(self):
         """ Add information about iap enrich to perform """

--- a/addons/partner_autocomplete/models/res_company.py
+++ b/addons/partner_autocomplete/models/res_company.py
@@ -15,7 +15,7 @@ COMPANY_AC_TIMEOUT = 5
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     partner_gid = fields.Integer('Company database ID', related="partner_id.partner_gid", inverse="_inverse_partner_gid", store=True)
     iap_enrich_auto_done = fields.Boolean('Enrich Done')

--- a/addons/partner_autocomplete/models/res_config_settings.py
+++ b/addons/partner_autocomplete/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     partner_autocomplete_insufficient_credit = fields.Boolean('Insufficient credit', compute="_compute_partner_autocomplete_insufficient_credit")
 

--- a/addons/partner_autocomplete/models/res_partner.py
+++ b/addons/partner_autocomplete/models/res_partner.py
@@ -26,7 +26,7 @@ VAT_COUNTRY_MAPPING = {
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     partner_gid = fields.Integer('Company database ID')
     additional_info = fields.Char('Additional info')

--- a/addons/partner_autocomplete/models/res_partner_autocomplete_sync.py
+++ b/addons/partner_autocomplete/models/res_partner_autocomplete_sync.py
@@ -8,6 +8,7 @@ _logger = logging.getLogger(__name__)
 
 
 class ResPartnerAutocompleteSync(models.Model):
+    _name = 'res.partner.autocomplete.sync'
     _description = 'Partner Autocomplete Sync'
 
     partner_id = fields.Many2one('res.partner', string="Partner", ondelete='cascade')
